File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/ui/playlist/Entry.java
package app.zxtune.ui.playlist;

import app.zxtune.TimeStamp;
import app.zxtune.core.Identifier;

// public for binding
public class Entry {
  public long id;
  public Identifier location;
  public String title;
  public String author;
  public TimeStamp duration;

  Entry(long id, Identifier location, String title, String author, TimeStamp duration) {
    this.id = id;
    this.location = location;
    this.title = title;
    this.author = author;
    this.duration = duration;
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/ui/playlist/Entry.kt
package app.zxtune.ui.playlist

import app.zxtune.TimeStamp
import app.zxtune.Util
import app.zxtune.core.Identifier

// public for binding
data class Entry(
    val id: Long,
    val location: Identifier,
    val title: String,
    val author: String,
    val duration: TimeStamp,
) {
    val displayTitle
        get() = Util.formatTrackTitle(title, location)
}
